fleet-terraform-generator/internal/module/generate.go

```diff
@@ -40,9 +40,12 @@ type Terraform struct {
 	File *terraform.File
 }
 
+var allowVarShadowing bool
```

Comment:
This global does not look like it is necessary.

Reply:
My golang experience is a bit rusty, I put this as a global so that it can be used without passing it through each function to where the variables shadowing check is.

https://github.com/andrewkroh/terraform-module-fleet/blob/a8e099977140fbf63a09401c4f20c68186e28597/fleet-terraform-generator/internal/module/generate.go#L287-L304

If there's a better way to do this let me know, i'll also see if this still works without it being a global var tomorrow.

Reply:
It looks like this does need to be a global var unless it's passed down through each function to expose it directly to the addVariable function. I originally started to do it that way but thought it was adding too many extra steps.